examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -1799,9 +1799,17 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
                     # Compute loss-weights as per Section 3.4 of https://arxiv.org/abs/2303.09556.
                     # Since we predict the noise instead of x_0, the original formulation is slightly changed.
                     # This is discussed in Section 4.2 of the same paper.
-                    snr = compute_snr(noise_scheduler, timesteps)
+
+                    if args.with_prior_preservation:
+                        # if we're using prior preservation, we calc snr for instance loss only -
+                        # and hence only need timesteps corresponding to instance images
+                        snr_timesteps, _ = torch.chunk(timesteps, 2, dim=0)
+                    else:
+                        snr_timesteps = timesteps
+
+                    snr = compute_snr(noise_scheduler, snr_timesteps)
                     base_weight = (
-                        torch.stack([snr, args.snr_gamma * torch.ones_like(timesteps)], dim=1).min(dim=1)[0] / snr
+                        torch.stack([snr, args.snr_gamma * torch.ones_like(snr_timesteps)], dim=1).min(dim=1)[0] / snr
```

Comment:
Shouldn't this affect the regular SDXL LoRA trainer too? Have we verified that? I am also not too sure if formulation-wise this is the correct thing to do i.e., when prior-preservation is done how should Min-SNR be computed. Your fix seems reasonable but I am honestly not very sure about the validity.

Reply:
> Shouldn't this affect the regular SDXL LoRA trainer too? Have we verified that?

I believe it does, since it's independent of the pivotal tuning training, that's why I pr'd the change for both trainers.

Reply:
Regarding the validity, I'm not very sure as well, my approach was in part based on the way we handled the use of both prior preservation loss and snr gamma in the [dreambooth training script](https://github.com/huggingface/diffusers/blob/4c483deb9077ef733d568ff0ae47233c35efd039/examples/dreambooth/train_dreambooth.py#L1280C17-L1305C1) 
where the snr gamma calculation is also seemingly computed only with regards to the instance loss (?). 
Maybe this sheds some light?

Reply:
It does. Thanks. Let's proceed with that, but also, verify some results with experiments?